paddleslim/dist/single_distiller.py

```diff
@@ -230,3 +230,76 @@ def loss(loss_func, program=None, **kwargs):
             func_parameters.setdefault(item[0], item[1])
     loss = loss_func(**func_parameters)
     return loss
+
+
+def _top_mask(x):
+    top_value, top_index = paddle.topk(x, 1)
+    return paddle.cast(x == top_value, "int32")
+
+
+def _cal_tc_nc_pred(x, top_mask):
+    """Calculate the predictions of target class and non-target class.
+    The predictions of target class is a binary distribution.
+    And the predictions of non-target class is softmax of  after removing target class.
```

Comment:
These sentences seem wired. How about this:

    """Calculate the predictions of target class and non-target class[es].
    The predictions of the target class is a binary distribution.
    And after removing the target class, the softmax on the remaining parts produces the non-target predictions.

Reply:
Fixed. Thx.